app/controllers/concerns/idv_session.rb

```diff
@@ -29,27 +28,32 @@ def confirm_phone_or_address_confirmed
   def idv_session
     @idv_session ||= Idv::Session.new(
       user_session: user_session,
-      current_user: effective_user,
+      current_user: idv_session_user,
       service_provider: current_sp,
     )
   end
 
   def idv_attempter_throttled?
     Throttle.new(
-      user: effective_user,
+      user: idv_session_user,
       throttle_type: :idv_resolution,
     ).throttled?
   end
 
-  def redirect_unless_effective_user
-    redirect_to root_url if !effective_user
+  def redirect_unless_idv_session_user
+    redirect_to root_url if !idv_session_user
   end
 
   def redirect_if_sp_context_needed
     return if sp_from_sp_session.present?
     return unless IdentityConfig.store.idv_sp_required
-    return if effective_user.profiles.any?
+    return if idv_session_user.profiles.any?
 
     redirect_to account_url
   end
+
+  def idv_session_user
+    # effective_user fallback usage for SessionErrorsController and CancellationsController
+    current_user || effective_user
```

Comment:
First, big 👍 for `idv_session_user` as a name here--it's clear what's going on.

I think we should remove this `effective_user` and provide a way for specific controllers to opt into being accessible by the document capture user:

```suggestion
  document_capture_user_id = session[:doc_capture_user_id]

  if document_capture_user_id
    # Allow controllers to opt in to be accessible by the doc capture user.
    if defined?(allow_document_capture_user) && allow_document_capture_user
      return User.find_by(id: document_capture_user_id)
    end
  end

  current_user
```

Controllers could then define an `allow_document_capture_user` method in addition to including the `IdvSession` concern.

Reply:
hi @matthinz  ... thanks for the review ... i get it ... remove effective user entirely.  Initially I did look at a similar approach.  However, the thing that deterred me was that I didn't see where the doc_capture_id gets removed from the session.  For instance, when a user completes the document capture process (ie: and is in a future step like verify_id or ssn etc.) does the session still contain the document_capture_id in the session?

Reply:
Theoretically, the doc_capture_user_id should only be present in the session _on the user's mobile device_. We validate the DocumentCaptureSession id and set the doc_capture_user_id in the [hybrid mobile entry controller](https://github.com/18F/identity-idp/blob/main/app/controllers/idv/hybrid_mobile/entry_controller.rb#L63). Once that's done the user is "logged in" for document capture until their Rails session times out.

Reply:
hi @matthinz 👋🏿  .... I have removed effective_user.  For general knowledge, I still have the question above about the life of the document_capture_user_id throughout the session?

Also, is there a need for a controller to opt in for using document_capture_user?